be/src/olap/rowset/segment_v2/segment_writer.h

```diff
@@ -120,11 +130,16 @@ class SegmentWriter {
     std::unique_ptr<MemTracker> _mem_tracker;
     uint32_t _row_count = 0;
 
-    vectorized::OlapBlockDataConvertor _olap_data_convertor;
+    std::unique_ptr<vectorized::OlapBlockDataConvertor> _olap_data_convertor;
     // used for building short key index or primary key index during vectorized write.
     std::vector<const KeyCoder*> _key_coders;
     std::vector<uint16_t> _key_index_size;
     size_t _short_key_row_pos = 0;
+
+    std::vector<uint32_t> _column_ids;
+    bool _has_key = true;
+    // written when add particial columns
+    uint32_t _num_rows_written = 0;
```

Comment:
The variable _num_rows_written is confused. It seems that num_rows_written works if users used segment writer in the order write keys, flush keys, writer colums? We should add more comment here.

// used to count rows to SegmentWriter, in vertical compaction, num_rows_written = actual_row_count * column groups number.  or we can find a better name.